README.md

```diff
@@ -49,23 +69,286 @@ oidc:
+
+
+### JwtValidationConfig
+[RFC 7519 - JSON Web Token (JWT)](https://datatracker.ietf.org/doc/html/rfc7519)
+| Parameter          | Type                                                      |
+|--------------------|-----------------------------------------------------------|
+| `validator`        | [`Validator`](#Validator) (defaults to [`Exist`](#Exist)) |
+| `require_expiry`   | Bool (defaults to `false`)                                |
+| `localpart_path`   | [`Path`](#Path) (optional)                                |
+| `user_id_path`     | [`Path`](#Path) (optional)                                |
+| `fq_uid_path`      | [`Path`](#Path) (optional)                                |
+| `displayname_path` | [`Path`](#Path) (optional)                                |
+| `required_scopes`  | Space separated string or a list of strings (optional)    |
+| `jwk_set`          | [JWKSet](https://datatracker.ietf.org/doc/html/rfc7517#section-5) or [JWK](https://datatracker.ietf.org/doc/html/rfc7517#section-4) (optional) |
+| `jwk_file`         | String (optional)                                         |
+
+Either `jwk_set` or `jwk_file` must be specified
+
+
+### IntrospectionValidationConfig
+[RFC 7662 - OAuth 2.0 Token Introspection](https://datatracker.ietf.org/doc/html/rfc7662)
+| Parameter          | Type                                                      |
+|--------------------|-----------------------------------------------------------|
+| `endpoint`         | String                                                    |
+| `validator`        | [`Validator`](#Validator) (defaults to [`Exist`](#Exist)) |
+| `auth`             | [`HttpAuth`](#HttpAuth) (optional)                        |
+| `localpart_path`   | [`Path`](#Path) (optional)                                |
+| `user_id_path`     | [`Path`](#Path) (optional)                                |
+| `fq_uid_path`      | [`Path`](#Path) (optional)                                |
+| `displayname_path` | [`Path`](#Path) (optional)                                |
+| `required_scopes`  | Space separated string or a list of strings (optional)    |
+
+Keep in mind, that default validator will always pass. According to the [spec](https://datatracker.ietf.org/doc/html/rfc7662), you probably want at least
+```yaml
+type: in
+path: 'active'
+validator:
+  type: equal
+  value: true
+```
+or
+```yaml
+['in', 'active', ['equal', true]]
+```
+
+### NotifyOnRegistration:
+| Parameter            | Type                               |
+|----------------------|------------------------------------|
+| `url`                | String                             |
+| `auth`               | [`HttpAuth`](#HttpAuth) (optional) |
+| `interrupt_on_error` | bool (defaults to `true`)          |
+
```

Comment:
```suggestion
### NotifyOnRegistration:
| Parameter            | Type                               |
|----------------------|------------------------------------|
| `url`                | String                             |
| `auth`               | [`HttpAuth`](#HttpAuth) (optional) |
| `interrupt_on_error` | bool (defaults to `true`)          |

The notify on registration endpoint must return status code 200 with an empty response `{}`.
```

The notify on registration only works when the endpoint returns the status code 200 with an empty response `{}`. Therefore, I added this to the documentation.

Otherwise the STA fails, here are examples of other responses:

Status code 200 with no response:
```
synapse-1   | 2024-06-28 09:07:23,510 - synapse.http.client - 426 - INFO - POST-20 - Received response to POST https://apimocha.com/8b4537b2/register: 200
synapse-1   | 2024-06-28 09:07:23,511 - synapse.handlers.auth - 2179 - WARNING - POST-20 - Failed to run module API callback <bound method TokenAuthenticator.check_oauth of <synapse_token_authenticator.token_authenticator.TokenAuthenticator object at 0xffff96641130>>: Expecting value: line 1 column 1 (char 0)
```
Status code 201 example:
```
synapse-1   | 2024-06-28 08:51:13,902 - synapse.http.client - 426 - INFO - POST-4 - Received response to POST https://apimocha.com/8b4537b2/register: 201
synapse-1   | 2024-06-28 08:51:13,904 - synapse.handlers.auth - 2179 - WARNING - POST-4 - Failed to run module API callback <bound method TokenAuthenticator.check_oauth of <synapse_token_authenticator.token_authenticator.TokenAuthenticator object at 0xffffb1690350>>: Expecting value: line 1 column 1 (char 0)
```
Status code 204 example:
```
synapse-1   | 2024-06-28 09:02:41,162 - synapse.http.client - 426 - INFO - POST-4 - Received response to POST https://apimocha.com/8b4537b2/register: 204
synapse-1   | 2024-06-28 09:02:41,162 - synapse.handlers.auth - 2179 - WARNING - POST-4 - Failed to run module API callback <bound method TokenAuthenticator.check_oauth of <synapse_token_authenticator.token_authenticator.TokenAuthenticator object at 0xffff96641130>>: Expecting value: line 1 column 1 (char 0)
```

Reply:
Added an exception for value error: [synapse_token_authenticator/token_authenticator.py:481](https://github.com/famedly/synapse-token-authenticator/blob/92b1ff2ee44ef3bbada23b85a89d5a3f542dcba6/synapse_token_authenticator/token_authenticator.py#L481-L482). It should not fail on non-json responses now. I hope it works as expected now

Reply:
Also, changed the order for notification and registration: now it notifies first and registers after that. This way, with `interrupt_on_error` it won't register and authenticate if notify call failed

Reply:
Sounds good! Thanks!